demangle-mode.el

```diff
@@ -53,28 +53,34 @@
 (require 'easymenu)
 (require 'tq)
 
-(eval-when-compile
-  (require 'rx))
-
 (defgroup demangle nil
   "Automatically demangle C++ symbols found in buffers."
   :group 'tools)
 
-(defcustom demangle-c++filt-executable "c++filt"
-  "The executable to use to decipher mangled symbols."
+(defcustom demangle-languages
+  '(("C++"
+     "\\(?:_Z\\|\\(?:_GLOBAL_[._$][DI]_\\)\\)[_$[:alnum:]]+"
+     ("c++filt" "--no-strip-underscore"))
+    ("D"
+     "_D[_[:alnum:]]+"
+     ("c++filt" "--no-strip-underscore" "--format=dlang")))
```

Comment:
This should be configurable to use `ddemangle` as an alternative.

Reply:
Update: We can always customize `demangle-languages` to use `ddemangle` so I guess this is a good default. Thanks!